app/controllers/active_storage/blobs/proxy_controller.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# Proxy files through application with authentication.
+# This avoids making the S3 bucket where files are stored public
+
+# Overwriting
+# https://github.com/rails/rails/blob/6-1-stable/activestorage/app/controllers/active_storage/blobs/proxy_controller.rb
+class ActiveStorage::Blobs::ProxyController < ActiveStorage::BaseController
+  include ActiveStorage::SetBlob
+  include ActiveStorage::SetHeaders
+  include ActionController::Live
+
+  before_action :authenticate_user!, :authorize_user!
```

Comment:
need to address 500 error when trying to download as an anonymous user

Reply:
This is a bug https://github.com/heartcombo/devise/issues/2332 that we will have to work-around.

Reply:
95d5b01

Reply:
verified this fix.